src/commonMain/kotlin/io/github/koalaplot/core/util/Gestures.kt

```diff
@@ -0,0 +1,220 @@
+package io.github.koalaplot.core.util
+
+import androidx.compose.foundation.gestures.awaitEachGesture
+import androidx.compose.foundation.gestures.awaitFirstDown
+import androidx.compose.foundation.gestures.calculateCentroid
+import androidx.compose.foundation.gestures.calculateCentroidSize
+import androidx.compose.foundation.gestures.calculatePan
+import androidx.compose.foundation.gestures.calculateZoom
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.input.pointer.PointerEvent
+import androidx.compose.ui.input.pointer.PointerInputScope
+import androidx.compose.ui.input.pointer.positionChanged
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.util.fastAny
+import androidx.compose.ui.util.fastForEach
+import kotlin.math.abs
+
+internal const val ZoomZeroPoint = 1.0f
+
+@Suppress("CyclomaticComplexMethod", "LongMethod") // expected
```

Comment:
It would be good to find a way to break this up instead of suppressing the checks. I noticed that it is implementing most things twice, once if lockZoomRatio is true and once if it is false. This could potentially be abstracted out to an interface with functions for all of the calculations, with 2 implementations, one for when the condition is true and one for when the condition is false, and then the logic check would be done once to instantiate the correct implementation.

Reply:
I isolated it into a separate interface and created two different implementations